src/visual/dataColor.js

```diff
@@ -53,10 +53,13 @@ export default function (seriesType) {
                 var singleDataColor = filteredIdx != null
                     && data.getItemVisual(filteredIdx, 'color', true);
 
-                if (!singleDataColor) {
-                    // FIXME Performance
-                    var itemModel = dataAll.getItemModel(rawIdx);
+                var singleDataBorderColor = filteredIdx != null
+                    && data.getItemVisual(filteredIdx, 'borderColor', true);
+
+                // FIXME Performance
+                var itemModel = dataAll.getItemModel(rawIdx);
```

Comment:
Since this sentence has performance issue in large data render cases,
we probably should not expose it out of the `if`.
It should better be executed only when necessary.